### PR TITLE
Upgrade Seata to 1.4.2.

### LINKED ISSUE #2543: [Seata]Update Seata to 1.4.2 in Spring Cloud Alibaba 2.2.x branch
我们鼓励使用英文，如果不能直接使用，可以使用翻译软件，您仍旧可以保留中文原文。
We recommend using English. If you are non-native English speaker, you can use the translation software.

Which Component
eg. Seata,

Is your feature request related to a problem? Please describe.
We are prepared to release a new version of Spring Cloud Alibaba in 2.2.x branch, we are plan to update some components.

Describe the solution you'd like
Update Seata to 1.4.2 in Spring Cloud Alibaba 2.2.x branch, and text the reliability.

### PR DESCRIPTION
**Pull request by misselvexu**
### Describe what this PR does / why we need it

Upgrade Seata to 1.4.2.

### Does this pull request fix one issue?

Fixes #2543 .

### Describe how you did it
- Upgrade Seata Version in `spring-cloud-alibaba-dependencies/pom.xml` .
- Improve Seata Module pom's dependencies.

### Describe how to verify it

I used the project's(https://github.com/alibaba/spring-cloud-alibaba/tree/master/spring-cloud-alibaba-examples/seata-example) examples to finish the compatibility issues after updating the version .

### Special notes for reviews
None.

### CODE REVIEW

**Comment by steverao** [comment]
Please add dingtalk account: steverao2021

**Comment by misselvexu** [comment]
> Please add dingtalk account: steverao2021

Requested .

File: spring-cloud-alibaba-starters/spring-cloud-starter-alibaba-seata/src/main/java/com/alibaba/cloud/seata/feign/SeataFeignObjectWrapper.java [hunk 1]
```diff
@@ -37,21 +34,12 @@ public class SeataFeignObjectWrapper {
 
 	private final BeanFactory beanFactory;
 
-	private CachingSpringLoadBalancerFactory cachingSpringLoadBalancerFactory;
```
Comment (steverao) [inline spring-cloud-alibaba-starters/spring-cloud-starter-alibaba-seata/src/main/java/com/alibaba/cloud/seata/feign/SeataFeignObjectWrapper.java]: Could you explain why you remove the code here and below?

Comment (misselvexu) [reply spring-cloud-alibaba-starters/spring-cloud-starter-alibaba-seata/src/main/java/com/alibaba/cloud/seata/feign/SeataFeignObjectWrapper.java]: Ohoo ! , after removing the code for testing, I forgot to restore it .

File: spring-cloud-alibaba-starters/spring-cloud-starter-alibaba-seata/src/main/java/com/alibaba/cloud/seata/rest/SeataRestTemplateAutoConfiguration.java [hunk 2]
```diff
@@ -58,4 +53,14 @@ public void init() {
 		}
 	}
 
+	@Configuration(proxyBeanMethods = false)
```
Comment (steverao) [inline spring-cloud-alibaba-starters/spring-cloud-starter-alibaba-seata/src/main/java/com/alibaba/cloud/seata/rest/SeataRestTemplateAutoConfiguration.java:56]: Why did you modify the codes here?

Comment (misselvexu) [reply spring-cloud-alibaba-starters/spring-cloud-starter-alibaba-seata/src/main/java/com/alibaba/cloud/seata/rest/SeataRestTemplateAutoConfiguration.java:56]: Consider guaranteeing the priority of `SeataRestTemplateInterceptor` initialization .

**Pull request merged**